astropy_timeseries/io/core.py

```diff
@@ -0,0 +1,109 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import warnings
+
+import numpy as np
+
+from astropy.io import registry, fits
+from astropy.table import Table
+from astropy.time import Time, TimeDelta
+
+from astropy_timeseries.sampled import TimeSeries
+
+__all__ = ["unified_reader"]
+
+
+def _unsupported(hdulist):
+    """
+    We checks to make sure we support these files.
+    Since KEPLER and TESS have the same checks, we can combine them here.
+    """
+
+    telescop = hdulist[0].header['telescop'].lower()
+    if telescop not in ["kepler", "tess"]:
+        raise NotImplementedError("{} is not implemented, only KEPLER or TESS are "
+                                  "supported through this reader".format(hdulist[0].header['telescop']))
+
+    # Get the lightcurve HDU
+    if telescop == 'tess':
+        hdu = hdulist['LIGHTCURVE']
+    if telescop == 'kepler':
+        hdu = hdulist[1]
+
+    if hdu.header['EXTVER'] > 1:
+        raise NotImplementedError("Support for {0} v{1} files not yet "
+                                  "implemented".format(hdu.header['TELESCOP'], hdu.header['EXTVER']))
+
+    # Check time scale
+    if hdu.header['TIMESYS'] != 'TDB':
+        raise NotImplementedError("Support for {0} time scale not yet "
+                                  "implemented in {1} reader".format(hdu.header['TIMESYS'], hdu.header['TELESCOP']))
+
+    # TODO: I wonder if we need these checks.
+    if telescop == 'kepler':
+        if "VERSION" in hdu.header.keys() and hdu.header["VERSION"] != '2.0':
+            warnings.warn("This is not a EVEREST pipeline version 2 KEPLER file.")
+        else:
+            warnings.warn("This seems to be a non EVEREST reduced KEPLER file.")
+
+    return hdu
+
+
+# TODO: Rename this
+def unified_reader(filename):
+    """
+    This serves as the basis for the FITS reader within astropy-timeseries.
```

Comment:
This is again a little too generic - it is a reader that supports Kepler and TESS formats.

Reply:
Renamed. Probably needs a better name.